### PR TITLE
getvalues refactor

### LINKED ISSUE #722: getvalues refactor
## Description
bring soca up to date with the pending changes in oops/ufo `feature/gevalues` branches

### PR DESCRIPTION
**Pull request by travissluka**
## Description
keeping up with the getvalues refactor in oops. Supposedly hofx will run faster now (yet to be tested with soca)

(and.... wooo, a net reduction of 301 lines of soca code!)

closes #722

## Dependencies

Requires branches with matching names in oops, saber, ufo

- waiting on https://github.com/JCSDA-internal/oops/pull/1680
- waiting on https://github.com/JCSDA-internal/saber/pull/193
- waiting on https://github.com/JCSDA-internal/ufo/pull/1930

### CODE REVIEW

File: test/testref/hofx_3d.test [hunk 1]
```diff
@@ -16,15 +16,15 @@ Test     :      lw   min=    0.000000   max=   88.036087   mean=   31.395529
...
-Test     : SeaSurfaceSalinity nobs= 100 Min=31.245713, Max=37.087544, RMS=34.753009
-Test     : ADT nobs= 100 Min=-1.250816, Max=1.421506, RMS=0.757217
-Test     : InsituTemperature nobs= 218 Min=7.832734, Max=30.431111, RMS=21.287619
-Test     : InsituSalinity nobs= 218 Min=34.160244, Max=35.992907, RMS=35.036438
-Test     : SeaIceFraction nobs= 100 Min=0.000000, Max=1.000000, RMS=0.694504
-Test     : SeaIceFreeboard nobs= 376 Min=-0.071509, Max=0.255906, RMS=0.054232
-Test     : Chlorophyll nobs= 137 Min=0.001273, Max=2.410599, RMS=0.357860
-Test     : SurfaceU nobs= 201 Min=-0.668706, Max=0.342825, RMS=0.176726
-Test     : SurfaceV nobs= 201 Min=-0.345640, Max=0.399404, RMS=0.114036
+Test     : CoolSkin nobs= 201 Min=-69.356342, Max=25.085506, RMS=21.135367
```
Comment (guillaumevernieres) [inline test/testref/hofx_3d.test:19]: Brrrr :cold_face:

Comment (travissluka) [reply test/testref/hofx_3d.test:19]: yeah, I noticed that. There are small differences in the interpolation, and it seems there is one point somewhere causing weird coolskin values. I think I convinced myself that masked/unmasked interpolation is indeed working correctly... so we'll just ignore that until a day ever comes that we actually use cool skin.

**Comment by travissluka** [comment]
FYI, the travisCI tests are all passing, but for some reason it times out when trying to upload the results.

**Review by guillaumevernieres — APPROVED** [review]

**Review by kbhargava — APPROVED** [review]
Looks good. Tests pass on orion.

**Comment by ytremolet** [comment]
OOPS PR has been merged.

**Pull request merged**